pennylane/ops/functions/equal.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.equal function.
+"""
+# pylint: disable=too-many-arguments
+import pennylane as qml
+
+
+def equal(op1, op2, check_interface=True, check_trainability=True, rtol=1e-5, atol=1e-9):
+    r"""Function for determining operator equality.
+
+    Args:
+        op1 (.Operator): First operator to compare
+        op2 (.Operator): Second operator to compare
+        check_interface (bool, optional): Whether to compare interfaces. Default: `True`
+        check_trainability (bool, optional): Whether to compare trainability status. Default: `True`
+        rtol (float, optional): Relative tolerance for parameters
+        atol (float, optional): Absolute tolerance for parameters
+
+    Returns:
+        bool: `True` if the operators are equal, else `False`
+
+    **Example**
+
+    Given two operators, ``qml.equal`` determines their equality:
+
+    >>> op1 = qml.RX(np.array(.12), wires=0)
+    >>> op2 = qml.RY(np.array(1.23), wires=0)
+    >>> qml.equal(op1, op1), qml.equal(op1, op2)
+    True False
+
+    .. details::
+        :title: Usage Details
+
+        You can use the optional arguments to get more specific results.
+
+        Consider the following comparisons:
+
+        >>> op1 = qml.RX(torch.tensor(1.2), wires=0)
+        >>> op2 = qml.RX(jax.numpy.array(1.2), wires=0)
+        >>> qml.equal(op1, op2)
+        False
+
+        >>> qml.equal(op1, op2, check_interface=False, check_trainability=False)
+        True
+
+        >>> op3 = qml.RX(np.array(1.2, requires_grad=True), wires=0)
+        >>> op4 = qml.RX(np.array(1.2, requires_grad=False), wires=0)
+        >>> qml.equal(op3, op4)
+        False
+
+        >>> qml.equal(op3, op4, check_trainability=False)
+        True
+
+        If the parameters do not have a `requires_grad` attribute, the function will return `False` if `check_trainability`
+        is `True`.
+    """
+    try:
```

Comment:
I'm not sure if `assert` should be used in this way. In general it is meant to catch things that should never happen. Maybe replace this with return statements

Reply:
done